test/mitmproxy/addons/test_proxyauth.py

```diff
@@ -109,6 +121,24 @@ def test_check():
         )
         assert not up.check(f)
 
+        ctx.configure(
```

Comment:
Ok so I looked at the mock option a bit and it is possible to have a mock connection, however it needs to be hard coded in the arguments of the Connection(), I can't just emulate a server and test  what I implemented against a fake server